### PR TITLE
Cleanup of key/values for cloning workflows into reqmgr2

### PR DESCRIPTION
**Pull request by amaltaro**
@AndrewLevin is having problems with clones into reqmgr2.
We have much more information in the workload of reqmgr2 requests, and this script was creating "new" workflows with some keys/values that are not meant to be provided by the user.

Andrew, @prozober please test it
@ticoann @vlimant please review

### CODE REVIEW

File: resubmit.py [hunk 1]
```diff
@@ -47,41 +45,31 @@ def modifySchema(helper, workflow, user, group, cache, events, firstLumi, backfi
...
-            result['RequestSizeEvents'] = value
-        #Set memory
-        elif key == 'Memory':
-            if memory != None:
-                result['Memory'] = memory
-            else:
-                result['Memory'] = value
-        #requestor info
+    for key, value in helper.data.request.schema.dictionary_whole_tree_().items():
+        if not value or key in paramBlacklist:
```
Comment (amaltaro) [inline resubmit.py]: including this would be a problem if value is False... I'll update the code soon

**Comment by amaltaro** [comment]
Code updated. Andrew, please create your clones with this patch. You may add that pprint back to your local script and send back the dictionary just before its injection, just for double check.

File: resubmit.py [hunk 2]
```diff
@@ -47,41 +45,31 @@ def modifySchema(helper, workflow, user, group, cache, events, firstLumi, backfi
...
     """
+    # list of keys to be removed from the new request schema
+    paramBlacklist = ['AcquisitionEra', 'BlockCloseMaxEvents', 'BlockCloseMaxFiles', 'BlockCloseMaxSize',
+                      'BlockCloseMaxWaitTime', 'CouchURL', 'CouchWorkloadDBName', 'CustodialGroup',
+                      'CustodialSubType', 'Dashboard', 'GracePeriod', 'Group', 'HardTimeout', 'InitialPriority',
+                      'inputMode', 'MaxMergeEvents', 'MaxMergeSize', 'MaxRSS', 'MaxVSize', 'MinMergeSize',
+                      'NonCustodialGroup', 'NonCustodialSubType', 'ProcessingString', 'OutputDatasets',
+                      'ReqMgr2Only', 'RequestDate' 'RequestorDN', 'RequestName', 'RequestStatus',
+                      'RequestTransition', 'RequestWorkflow', 'SiteWhitelist', 'SoftTimeout',
+                      'SoftwareVersions', 'SubscriptionPriority', 'Team', 'timeStamp']
```
Comment (ticoann) [inline resubmit.py]: Hi Alan, other than below keys why others has to be removed?
'ReqMgr2Only', 'RequestDate' 'RequestorDN', 'RequestName', 'RequestStatus', 'RequestTransition', 'RequestWorkflow',  'Team', 'timeStamp'

Comment (amaltaro) [inline resubmit.py]: Most of them are parameters defined during request assignment. That's why I took them out here, however, there is clearly some keys that should remain... like AcquisitionEra and ProcessingString

**Comment by amaltaro** [comment]
Andrew had problems with the Group key, so I un-blacklisted it.

**Comment by amaltaro** [comment]
Let me make another fix to this PR

**Comment by vlimant** [comment]
@prozober : please report on a test here
@amaltaro : what is the symptom ? failing to submit a clone, or bad behavior of the clone ? 
https://github.com/CMSCompOps/WmAgentScripts/blob/master/Unified/rejector.py#L86 should also have some adjustments to superfluous parameter ?

**Comment by amaltaro** [comment]
Added `Group` back. I've just ran a test and it looks Ok.

Besides wrong information (RequestTransition for instance), I didn't spot any real issues. 
While we do not have this handling on the server side, I just wanted to POST requests as clean as possible and isolate what is creation from what is assignment parameters.

**Comment by prozober** [comment]
I made some tests sending workflows to cmsweb.cern.ch and everything is OK. But, when I send clones to cmsweb-testbed.cern.ch I am getting [1], am I doing something wrong?

`python resubmit.py -b -v --testbed prozober_task_HIG-RunIISpring16DR80-01435__v1_T_160711_152005_6280 `

[1]
> <p>Create request failed, (400, 'Error in Workload Validation: Failure to load ConfigCache while validating workload: ConfigCacheException\nMessage: Document with id de94861ef8ed7510601c7bbedd807cc6 not found in couch\nCouchNotFoundError - reason: Object Not Found, data: {} result: NoneTraceback (most recent call last):\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Cache/WMConfigCache.py", line 279, in loadByID\n    self.document = self.database.document(id = configID)\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 303, in document\n    return Document(id=id, inputDict=self.get(uri))\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 107, in get\n    encode, decode, contentType)\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 122, in makeRequest\n    getattr(e, "reason", None), data)\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 139, in checkForCouchError\n    raise CouchNotFoundError(reason, data, result)\nCouchNotFoundError: CouchNotFoundError - reason: Object Not Found, data: {} result: None\n\n\tModuleName : WMCore.Cache.WMConfigCache\n\tMethodName : loadByID\n\tClassInstance : None\n\tFileName : /data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Cache/WMConfigCache.py\n\tClassName : None\n\tLineNumber : 292\n\tErrorNr : 0\n\nTraceback: \n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Cache/WMConfigCache.py", line 279, in loadByID\n    self.document = self.database.document(id = configID)\n\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 303, in document\n    return Document(id=id, inputDict=self.get(uri))\n\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 107, in get\n    encode, decode, contentType)\n\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 122, in makeRequest\n    getattr(e, "reason", None), data)\n\n  File "/data/srv/beHG1607i-comp4/sw/slc6_amd64_gcc493/cms/reqmgr/1.0.18.pre6/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 139, in checkForCouchError\n    raise CouchNotFoundError(reason, data, result)\n\n')</p>

**Comment by amaltaro** [comment]
Paola, I believe it was not working before as well. Can you x-check it please?
Otherwise, we need to explicitly set 'ConfigCacheUrl' key/value

**Comment by prozober** [comment]
Everything is working now, thanks Alan!

**Comment by amaltaro** [comment]
Andrew is/was having problems with the AAA flags. So I just removed them from the clone parameters object and we let it to be set or not in the right place (assignment).

@AndrewLevin please test this change (if your assignment script is still hard-coding TrustSitelists to True, then it's a hopeless case and there is nothing else that I can do in a short term...)

**Comment by vlimant** [comment]
@AndrewLevin please update if alright

**Comment by AndrewLevin** [comment]
I think it is alright now

**Pull request merged**